Uchu.World/Objects/Components/Player/ModularBuilderComponent.cs

```diff
@@ -153,6 +163,53 @@ public async Task ConfirmFinish()
             IsBuilding = false;
         }
 
+        private async Task CreateNewModel(Lot[] parts)
+        {
+            var inventory = GameObject.GetComponent<InventoryManagerComponent>();
+
+            var model = new LegoDataDictionary
+            {
+                ["assemblyPartLOTs"] = LegoDataList.FromEnumerable(parts.Select(s => s.Id))
+            };
+
+            Lot modelLot = GetModelLotForBuildMode(Mode);
+            await inventory.AddLotAsync(modelLot, 1, model, InventoryType.Models);
+        }
+
+        private async Task GetBackOldModel()
+        {
+            if (CurrentModel == null)
+                return;
+
+            var inventory = GameObject.GetComponent<InventoryManagerComponent>();
+
+            var parts = new Item[CurrentModel.Length];
+            var model = CurrentModel;
+
+            // Remove model so it doesn't get duplicated
+            CurrentModel = null;
+
+            // Check if all parts are still available in the players inventory
+            for (var i = 0; i < parts.Length; i++)
+            {
+                Lot lot = model[i];
+
+                var item = inventory.FindItem(lot, InventoryType.TemporaryModels);
+                if (item == null)
+                    item = inventory.FindItem(lot, InventoryType.Models);
```

Comment:
You can replace the if statement with just a `??`:

```suggestion
                var item = inventory.FindItem(lot, InventoryType.TemporaryModels) 
                           ?? inventory.FindItem(lot, InventoryType.Models);
```